File: Data Structures & Algorithms/string-encode-and-decode/submission-0.cpp
class Solution {
public:

    string encode(vector<string>& strs) {
        string code = "";
        for (auto str : strs){
            code += str + "\n";
        }

        return code;
    }

    vector<string> decode(string s) {
        string word=""; 
        vector<string> ans;
        for (auto v : s){
            string value(1, v);
            if (value == "\n"){
                ans.push_back(word);
                word = "";
                continue;
            }
            word += value;
        }

        return ans;
    }
};

File: Data Structures & Algorithms/permutation-string/submission-0.py
class Solution:
    def checkInclusion(self, s1: str, s2: str) -> bool:
        count = {}

        for l in s1:
            count[l] = 1 + count.get(l, 0)
        
        k = len(s1)
        j = len(s2)

        if j < k:
            return False
        
        f = 0
        r = 0
        ans = {}
        while r < j:
            print
            ans[s2[r]] = 1 + ans.get(s2[r], 0)
            if r - f + 1 >= k:
                if ans == count:
                    return True

                ans[s2[f]] -= 1
                if ans[s2[f]] == 0:
                    ans.pop(s2[f])
                f += 1
            r += 1
        
        return False
        
